bin/road/main.rs

```diff
@@ -41,3 +49,304 @@ fn main() {
 
     harness.main_loop(game);
 }
+
+#[cfg(target_arch = "wasm32")]
+fn main() {
+    use env_logger::{Builder};
+    use log::{LevelFilter};
+
+    console_error_panic_hook::set_once();
+
+    Builder::new()
+        .format(|_buf, record| {
+            let message = format!("{}: {}", record.level(), record.args());
+            log(&message);
+            Ok(())
+        })
+        .filter(None, LevelFilter::Info)
+        .init();
+
+    create_fs();
+
+    async fn run() {
+        let (harness, settings) = boilerplate::Harness::init_async(boilerplate::HarnessOptions {
+            title: "road",
+            uses_level: true,
+        }).await;
+
+        let game = game::Game::new(
+            &settings,
+            harness.color_format,
+            harness.extent,
+            &harness.device,
+            &harness.queue,
+            &harness.downlevel_caps,
+        );
+
+        harness.main_loop(game);
+    }
+
+    wasm_bindgen_futures::spawn_local(run());
+}
+
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console, js_name = log)]
+    fn log(a: &str);
+}
+
+#[cfg(target_arch = "wasm32")]
+fn create_file(file: &str, buf: &[u8]) {
+    use std::{fs::File, io::Write};
+
+    info!("Creating file {}", file);
+    File::create(file).expect(&format!("Unable to create {}", file))
+        .write(buf)
+        .expect(&format!("Unable to write in {}", file));
+}
+
+#[cfg(target_arch = "wasm32")]
+fn create_fs() {
+    info!("Creating fs");
+
+	// vangers-rs resources
+	create_file("res/shader/quat.inc.glsl", include_bytes!("../../res/shader/quat.inc.glsl"));
```

Comment:
there must be a way to just say "I want this folder", mustn't it?

Reply:
This API exists in emscripten, but not in wasm32-unknown-unknown. I doing short article about how it works in wasm32-unknown-unknown, hope it will be ready asap.